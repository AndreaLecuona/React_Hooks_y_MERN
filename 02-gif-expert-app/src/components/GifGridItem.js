import React from 'react';
import Proptypes from "prop-types";

export const GifGridItem = ( { id, title, url } ) => {

    // console.log(id, title, url);

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    )
};

GifGridItem.proptypes = {
    title: Proptypes.string.isRequired,
    url: Proptypes.string.isRequired
};
