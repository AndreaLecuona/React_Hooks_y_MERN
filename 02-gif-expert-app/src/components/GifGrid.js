import React from 'react';
import Proptypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( { categoria } ) => {

    // const [images, setImages] = useState([]);
    
    const { data: images, loading } = useFetchGifs( categoria );

    // useEffect( () => {
    //     getGifs( categoria ).then( imgs => setImages( imgs )); //al modularizar getGifs, hubo que convertirlo en async y por ende en promesa
    // }, [ categoria ]); //dependencia a la categoria: si cambia la categoria vuelve a hacer la peticion a la api

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">"{ categoria }"</h3>

            { loading && <p className="animate__animated animate__flash">Cargando</p> }
            
            <div className="card-grid">
                {
                    images.map( img =>
                        <GifGridItem 
                            key={img.id}

                            // img={ img }
                            { ...img }
                        />
                    )
                }
            </div>
        </>
    )
};

GifGrid.proptypes = {
    categoria: Proptypes.string.isRequired
};

// SIN DESTRUCTURING

// <div>
// <h3>{ category }</h3>
// <ol>
//     {
//         images.map( image =>
//             <li key={image.id}>{image.title}</li>)
//     }
// </ol>
// </div
