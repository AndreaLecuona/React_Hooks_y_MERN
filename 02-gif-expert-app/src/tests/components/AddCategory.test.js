import React from 'react';
import { AddCategory } from "../../components/AddCategory";
import {shallow} from 'enzyme';


describe('Pruebas en el componente AddCategory', () => {

    const setCategories = jest.fn();
    
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories }/> );
    });

    test('Debe renderizarse correctamente', () => {    

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe modificar el texto del input', () => {
        
        const input = wrapper.find('input');

        const value = 'Hola Mundo';

        input.simulate('change', {
            target: {
                value: value
            }
        });

        //opcion2 con un <p> en el componente
        //expect(wrapper.find('p').text().trim()).toBe( value );

    });
    
    test('No debe hacer submit automÃ¡ticamente', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });

        expect( setCategories ).not.toHaveBeenCalled();

    });

    test('Debe llamar a setCategories y limpiar el input cuado hace submit', () => {
        
        const value = 'Hola Mundo';
        const input = wrapper.find('input');

        input.simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', { preventDefault(){} });
        
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes( 1 );
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) ); //que sea una funcion y no un number o string

        expect( input.prop('value') ).toBe( '' );

    });
    
    
    
});