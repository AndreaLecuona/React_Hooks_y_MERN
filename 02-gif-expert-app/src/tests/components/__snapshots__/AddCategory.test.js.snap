// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory Debe renderizarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Type something..."
    type="text"
    value=""
  />
</form>
`;
